src/tool/GenerateAction.cpp

```diff
@@ -13,15 +13,108 @@
 #include "lib/Lib/AbsoluteCompilationDatabase.hpp"
 #include "lib/Lib/ConfigImpl.hpp"
 #include "lib/Lib/CorpusImpl.hpp"
+#include "llvm/Support/Program.h"
 #include <mrdocs/Generators.hpp>
 #include <mrdocs/Support/Error.hpp>
 #include <mrdocs/Support/Path.hpp>
 #include <clang/Tooling/JSONCompilationDatabase.h>
+
 #include <cstdlib>
+#include <unordered_map>
 
 namespace clang {
 namespace mrdocs {
 
```

Comment:
I'm under the impression that this is growing too much and we're breaking the separation of concerns here. Maybe all the functionality related to compilers can go into new .h/.cpp files under src/lib/Support. `Compiler.h/.cpp`?

Reply:
Done